### PR TITLE
[dataframe] Upgrading C++ DataFrame to version 2.3.0

### PR DESCRIPTION
**Pull request by hosseinmoein**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by hosseinmoein** [comment]
It seems that C++23 is not supported in the majority of your platforms yet? I just added them to the `ci.baseline.txt` file

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -1369,3 +1369,14 @@ vcpkg-ci-wxwidgets:x64-windows-static-md=pass
...
+dataframe:arm64-android=fail
+dataframe:arm64-windows=fail
+dataframe:arm-neon-android=fail
+dataframe:x64-android=fail
+# dataframe:x64-linux=fail
+# dataframe:x64-osx=fail
+# dataframe:x64-windows=fail
+# dataframe:x64-windows-static=fail
+# dataframe:x64-windows-static-md=fail
+# dataframe:x86-windows=fail
```
Comment (MonicaLiu0311) [inline scripts/ci.baseline.txt:1382]: Remove unnecessary sentences.

Comment (MonicaLiu0311) [inline scripts/ci.baseline.txt:1382]:
```suggestion
```
Remove unnecessary sentences.

Comment (hosseinmoein) [reply scripts/ci.baseline.txt]: @MonicaLiu0311 ,
Done

**Pull request merged**